factionRenderer.py

```diff
@@ -1,105 +1,116 @@
-]
-mechparts = [
-    {"item": "[[JagerMech II#JM7-D|JagerMech II JM7-D]]</br>"},
-    {"item": "[[Valkyrie II#VLK-II-2A|Valkyrie II VLK-II-2A]]</br>"}
-]
-vehicles = [
-    {"item": "[[Paladin#Paladin Prototype Defense System|Paladin Prototype Defense System]]</br>"},
-]
-mechparts = [
-    {"item": "[[JagerMech II#JM7-D|JagerMech II JM7-D]]</br>"},
-    {"item": "[[Valkyrie II#VLK-II-2A|Valkyrie II VLK-II-2A]]</br>"}
-]
-battlearmors = [
-    {"item": "[[Grenadier Battle Armor#|Grenadier Battle Armor]]</br>"},
-    {"item": "[[Infiltrator Mk II Battle Armor#|Infiltrator Mk II Battle Armor]]</br>"}
-]
-contracts = [
-    {"item": "[[Contracts#Contract for Challenger Tank Airdrop|Contract for Challenger Tank Airdrop]]</br>"},
-    {"item": "[[Contracts#Contract for Templar Airdrop|Contract for Templar Airdrop]]</br>"},
-    {"item": "[[Contracts#Permanent Contract for Grenadier BA Airdrop|Permanent Contract for Grenadier BA Airdrop]]</br>"},
-    {"item": "[[Contracts#Permanent Contract for Infiltrator Mk II BA Airdrop|Permanent Contract for Infiltrator Mk II BA Airdrop]]</br>"}
-]
 environment = Environment(loader=FileSystemLoader("templates/"))
 template = environment.get_template("factionStore.tpl")
 
-results_filename = "factionTable.wiki"
-results_template = environment.get_template("factionStore.tpl")
-context = {
+example_dict =  {'Rasalhague.json': ['Weapon_Autocannon_LB10XAC',
+                     'Weapon_PPC_PPC_SNUBNOSE',
+                     'Weapon_Autocannon_UAC5_0-STOCK',
+                     'Weapon_SRM_SSRM2_0-STOCK',
+                     'Ammo_AmmunitionBox_LBX_CLUSTER_AC10',
+                     'Ammo_AmmunitionBox_LBX_SLUG_AC10',
+                     'Ammo_AmmunitionBox_Ultra_AC5',
+                     'Ammo_AmmunitionBox_Streak_SRM',
+                     'mechdef_viking_VKG-2F',
+                     'mechdef_vonrohrs_VON-4RH-5',
+                     'mechdef_vonrohrs_VON-4RH-6',
+                     'vehicledef_REDKITE',
+                     'vehicledef_UNNSVIN',
+                     'vehicledef_UNNSVIN_BALLISTIC',
+                     'vehicledef_UNNSVIN_MISSILE',
+                     'vehicledef_TAGDRONE',
+                     'vehicledef_VTOLDRONE',
+                     'mechdef_ba_krupp',
+                     'Weapon_Medium_Recoilless_BA',
+                     'Gear_Attachment_AutocannonFCS',
+                     'Gear_Attachment_PPCFCS',
+                     'Gear_Contract_Mech_Gunslinger_FRR',
+                     'Gear_Contract_BattleArmor_Krupp_Rasalhague']}
+
+codebase_dir = "/home/eadderley/fungit/BattleTech-Advanced"
+
+def render_factionstore(faction, items):
+    faction_name = faction[:-5]
+    results_filename = faction_name+"_store_Table.wiki"
+    faction_info=get_faction_specific_info(faction_name)
+    weapons = []
+    ammunitions = []
+    gears = []
+    mechs = []
+    vehicles = []
+    battlearmors = []
+    contracts = []
+    for item in items:
```

Comment:
this is a good start - because you've defined a similar step for each item, we can easily refactor this into an object based approach if you'd like. The assumption there is that eventually we might want to do different things on different types of objects, or that we might be adding more objects in the future. 

The current structure means the code has to change in multiple places. As currently written this isn't terrible, this code is easy to understand because it is all in one place. That is the trade-off; if we refactor it we are increasing the layers of indirection to make the smaller pieces more digestible and maintainable at the cost of potentially increasing the cost of understanding the whole program at once.